xcodeproj/internal/xcodeproj_macro.bzl

```diff
@@ -287,6 +288,18 @@ def xcodeproj(
               custom schemes are provided.
             </li>
             </ul>
+        scheme_autogeneration_config: Optional. A struct returned by
+            `xcschemes.autogeneration_config`.
+
+            Allows further configuration of `scheme_autogeneration_mode`.
+
+            Available configurations:
+
+            <ul>
+            <li>
+              `xcschemes.autogeneration_exclude_patterns`: Used to skip creating schemes matching provided regex patterns.
```

Comment:
We should be clearer here that this only applies to automatic scheme generation

Reply:
A few lines above I added
```
Allows further configuration of `scheme_autogeneration_mode`.
```
plus I believe the name `xcschemes.autogeneration_exclude_patterns` conveys both "scheme" and "autogeneration"? 

I don't mind making it clearer if the above is not enough. Mind sharing one example of how you'd change the doc?

Reply:
i think that callout is good! what happens right now if someone provides `"none"` for auto generation but also supplies this attribute? Do we fail?

Reply:
We don't fail atm, but yeah maybe we should. Otherwise (regardless if the behavior is correct or not) users might think it's doing something. I can update that and then be more specific in the doc about this.

Reply:
I guess one question here is if we want exclude patterns to apply if custom schemes are being used? It makes sense to apply it in `all` and `auto with no custom schemes`, but I guess we should agree on what to do with `auto with custom schemes`?

Reply:
id say it shouldnt apply to custom schemes, if you are supplying a custom scheme you already have control over what targets are added to it.

Reply:
SGTM, seems aligned with the behaviour of  `auto with custom schemes` where nothing else is created automatically giving control to the user. So I'll make it so these exclusion patterns apply on `all` and `auto without custom schemes` only.

Reply:
https://github.com/MobileNativeFoundation/rules_xcodeproj/pull/3027/commits/0610f4c9d63c4995c1e81e13f545757816f00041

Let me know if this addresses your suggestion!